#  Exercise-3

Let's run a regression analysis and interpret the result using `wine` dataset.

Let's first import libraries and the dataset. 

In [3]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from sklearn import linear_model
from sklearn.linear_model import LinearRegression 
import statsmodels.api as sm
import warnings
warnings.filterwarnings('ignore')

In [4]:
wine_white=pd.read_csv('datasets/winequality-white.csv',sep=';')
wine_red=pd.read_csv('datasets/winequality-red.csv',sep=';')

In the following code, I combine the two dataset along the rows so that I have just one comprehensive dataset instead of two separate ones.

In [5]:
combined=wine_white.append(wine_red)

In [4]:
combined.head()

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.0              0.27         0.36            20.7      0.045   
1            6.3              0.30         0.34             1.6      0.049   
2            8.1              0.28         0.40             6.9      0.050   
3            7.2              0.23         0.32             8.5      0.058   
4            7.2              0.23         0.32             8.5      0.058   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 45.0                 170.0   1.0010  3.00       0.45   
1                 14.0                 132.0   0.9940  3.30       0.49   
2                 30.0                  97.0   0.9951  3.26       0.44   
3                 47.0                 186.0   0.9956  3.19       0.40   
4                 47.0                 186.0   0.9956  3.19       0.40   

   alcohol  quality  
0      8.8        6  
1      9.5        6  
2     10.1        6  
3      9.9        6  
4      9.9        6

Our independent variables include all variables but `quality`, which is the dependent variable. There are several ways to sort out the `quality` variable from the other bunch of variables. Here, I prefer using `!` which filters the dataset so as to include all but `quality` variable. Then, by convention, I assign all the independent variables to X data frame and dependent variable to Y data frame.

In [5]:
X=combined.loc[:,combined.columns!='quality']

In [6]:
y=combined['quality']

In [7]:
X = sm.add_constant(X)
sm_ols=sm.OLS(y,X).fit()

In [8]:
sm_ols.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                quality   R-squared:                       0.292
Model:                            OLS   Adj. R-squared:                  0.291
Method:                 Least Squares   F-statistic:                     243.3
Date:                Sun, 02 Aug 2020   Prob (F-statistic):               0.00
Time:                        00:23:30   Log-Likelihood:                -7215.5
No. Observations:                6497   AIC:                         1.445e+04
Df Residuals:                    6485   BIC:                         1.454e+04
Df Model:                          11                                         
Covariance Type:            nonrobust                                         
========================================================================================
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                   55.7627     11.894      4.688      0.000      32.447      79.079
fixed acidity            0.0677      0.016      4.346      0.000       0.037       0.098
volatile acidity        -1.3279      0.077    -17.162      0.000      -1.480      -1.176
citric acid             -0.1097      0.080     -1.377      0.168      -0.266       0.046
residual sugar           0.0436      0.005      8.449      0.000       0.033       0.054
chlorides               -0.4837      0.333     -1.454      0.146      -1.136       0.168
free sulfur dioxide      0.0060      0.001      7.948      0.000       0.004       0.007
total sulfur dioxide    -0.0025      0.000     -8.969      0.000      -0.003      -0.002
density                -54.9669     12.137     -4.529      0.000     -78.760     -31.173
pH                       0.4393      0.090      4.861      0.000       0.262       0.616
sulphates                0.7683      0.076     10.092      0.000       0.619       0.917
alcohol                  0.2670      0.017     15.963      0.000       0.234       0.300
==============================================================================
Omnibus:                      144.075   Durbin-Watson:                   1.646
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              324.712
Skew:                          -0.006   Prob(JB):                     3.09e-71
Kurtosis:                       4.095   Cond. No.                     2.49e+05
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 2.49e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

The estimated model reveals the following equation:

    quality=55.7627+0.0677fixed acidity-1.3279volatile acidity-0.1097citric acid+0.0436residual sugar-0.4837chlorides+0.0060free sulfur dioxide-0.0025total sulfur dioxide-54.9669density+0.4393pH+0.7683sulphates+0.2670alcohol	